### PR TITLE
[OTE-823] Fix FNS onchain events staging + retrieval logic

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
Fix FNS onchain events staging + retrieval logic. 

### Test Plan
Tested by running FNS + load-testing locally. 

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method `Precommit` for enhanced ABCI Precommit logic in the clob module.
	- Updated streaming manager to utilize a new codec for improved event serialization and deserialization.
  
- **Bug Fixes**
	- Removed restrictions allowing gRPC streaming to work with optimistic execution enabled.

- **Tests**
	- Updated test cases to reflect changes in flag validation, allowing optimistic execution with gRPC streaming.

### CODE REVIEW

File: protocol/app/flags/flags.go [hunk 1]
```diff
@@ -163,10 +163,6 @@ func (f *Flags) Validate() error {
 
 	// Grpc streaming
 	if f.GrpcStreamingEnabled {
-		if f.OptimisticExecutionEnabled {
```
Comment (teddyding) [inline protocol/app/flags/flags.go:166]: Previously added to prevent both flags from being accidentally set true

File: protocol/x/clob/module.go [hunk 2]
```diff
@@ -177,6 +177,16 @@ func (am AppModule) PreBlock(ctx context.Context) (appmodule.ResponsePreBlock, e
 	}, nil
 }
 
+// BeginBlock executes all ABCI BeginBlock logic respective to the clob module.
+func (am AppModule) Precommit(ctx context.Context) error {
```
Comment (teddyding) [inline protocol/x/clob/module.go:181]: Callback was not previously implemented, so `ClobModule.Precommit` was not being run

**Review by jayy04 — APPROVED** [review]

**Review by jonfung-dydx — APPROVED** [review]

File: protocol/app/flags/flags_test.go [hunk 3]
```diff
@@ -113,14 +113,17 @@ func TestValidate(t *testing.T) {
 				OptimisticExecutionEnabled: true,
 			},
 		},
-		"failure - optimistic execution cannot be enabled with gRPC streaming": {
+		"success - optimistic execution canbe  enabled with gRPC streaming": {
```
Comment (jonfung-dydx) [inline protocol/app/flags/flags_test.go:116]: typo

**Comment by teddyding** [comment]
@mergifyio backport release/protocol/v7.x

**Pull request merged**